test/integration/http2_integration_test.cc

```diff
@@ -85,6 +85,20 @@ TEST_P(Http2IntegrationTest, TwoRequests) { testTwoRequests(); }
 
 TEST_P(Http2IntegrationTest, Retry) { testRetry(); }
 
+TEST_P(Http2IntegrationTest, PerStreamIdleTimeoutBeforeUpstreamHeaders) {
```

Comment:
Optional but I'd like to move away from the "add each test to 3 files" paradigm towards using HttpProtocolIntegrationTest

If we're going to leave the timeouts 1s I'd encourage moving these to their own file, 
defining typedef HttpProtocolIntegrationTest IdleTimeoutsTest and just running all 4 without all the extra TEST_Ps.

Feel free to leave it here for consistency and I'll go split things out at some later point.